test/format/format_inline.h

```diff
@@ -281,11 +281,20 @@ static inline void
 wt_wrap_open_cursor(WT_SESSION *session, const char *uri, const char *config, WT_CURSOR **cursorp)
 {
     WT_DECL_RET;
+    bool is_checkpoint;
+
+    is_checkpoint = config != NULL && WT_PREFIX_MATCH(config, "checkpoint");
 
     /* WT_SESSION.open_cursor can return EBUSY if concurrent with a metadata operation, retry. */
     while ((ret = session->open_cursor(session, uri, NULL, config, cursorp)) == EBUSY)
         __wt_yield();
-    testutil_checkfmt(ret, "%s", uri);
+    /*
+     * WT_SESSION.open_cursor can also return ENOENT, this is fine in the case of a checkpoint
+     * cursor.
+     */
+    testutil_assertfmt(ret == 0 || (is_checkpoint && ret == ENOENT), "%s", uri);
+    if (is_checkpoint && ret == ENOENT && *cursorp != NULL)
+        testutil_check((*cursorp)->close(*cursorp));
```

Comment:
Can you check the call to `wt_wrap_open_cursor` in `table_dump_page`? This can also open a checkpoint cursor and currently assumes the cursor will always open successfully

Reply:
As discussed, the logic was moved inside `wt_wrap_open_cursor` to handle the failure. We no longer expect the caller to handle it.